junit5-extension/src/test/java/io/kroxylicious/testing/kafka/junit5ext/ParameterExtensionTest.java

```diff
@@ -126,167 +118,177 @@ public void producerConfiguration(KafkaCluster cluster,
-
-        ee = assertThrows(ExecutionException.class, () -> describeCluster(cluster.getKafkaClientConfiguration("eve", "quux")),
-                "Expect unknown user to throw");
-        assertInstanceOf(SaslAuthenticationException.class, ee.getCause());
+    void saslPlainAuthExplicitMechanism(@BrokerCluster @SaslMechanism("PLAIN") @User(user = "alice", password = "foo") KafkaCluster cluster) {
+        doAuthExpectSucceeds(cluster, "alice", "foo");
     }
 
     @Test
-    public void saslPlainAuthenticatingClusterParameter1User(
-                                                             @BrokerCluster @SaslPlainAuth(user = "alice", password = "foo") KafkaCluster cluster)
-            throws ExecutionException, InterruptedException {
-        var dc = describeCluster(cluster.getKafkaClientConfiguration("alice", "foo"));
-        assertEquals(1, dc.nodes().get().size());
-        assertEquals(cluster.getClusterId(), dc.clusterId().get());
-
-        var ee = assertThrows(ExecutionException.class, () -> describeCluster(cluster.getKafkaClientConfiguration("alice", "baz")),
-                "Expect bad password to throw");
-        assertInstanceOf(SaslAuthenticationException.class, ee.getCause());
-
-        ee = assertThrows(ExecutionException.class, () -> describeCluster(cluster.getKafkaClientConfiguration("bob", "bar")),
-                "Expect unknown user to throw");
-        assertInstanceOf(SaslAuthenticationException.class, ee.getCause());
+    void saslScramAuth(@BrokerCluster @SaslMechanism("SCRAM-SHA-256") @User(user = "alice", password = "foo") KafkaCluster cluster) {
+        doAuthExpectSucceeds(cluster, "alice", "foo");
     }
 
     @Test
-    public void tlsClusterParameter(
-                                    @Tls @BrokerCluster(numBrokers = 1) KafkaCluster cluster,
-                                    Admin admin)
-            throws ExecutionException, InterruptedException {
-        String bootstrapServer = cluster.getBootstrapServers();
-        assertFalse(bootstrapServer.contains(","), "expect a single bootstrap server");
+    void saslAuthWithManyUsers(@BrokerCluster @User(user = "alice", password = "foo") @User(user = "bob", password = "bar") KafkaCluster cluster) {
+        doAuthExpectSucceeds(cluster, "alice", "foo");
+        doAuthExpectSucceeds(cluster, "bob", "bar");
+    }
+
+    @Test
+    @SuppressWarnings("deprecated")
+    @Deprecated
+    void saslPlainAuthDeprecatedAnnotation(@BrokerCluster @User(user = "alice", password = "foo") KafkaCluster cluster) {
+        doAuthExpectSucceeds(cluster, "alice", "foo");
+    }
+
+    @Test
+    @SuppressWarnings({ "deprecated", "java:S4144" })
+    void saslPlainAuthDeprecatedAnnotationManyUsers(@BrokerCluster @SaslPlainAuth(user = "alice", password = "foo") @SaslPlainAuth(user = "bob", password = "bar") KafkaCluster cluster) {
+        doAuthExpectSucceeds(cluster, "alice", "foo");
+        doAuthExpectSucceeds(cluster, "bob", "bar");
+    }
```

Comment:
Should we also cover the `List` version of `SaslPlainAuth`?

Reply:
missing test added.